### ISSUE TITLE
Validate payment prior to creating schedule

### ISSUE THREAD

**Issue by jbaxleyiii** [NewSpring/Apollos#439]
Currently NMI does not allow this natively. We will need to create a `validation` transaction to run at confirm prior to creating a schedule

**Comment by jbaxleyiii** [comment]
Planned refactor / order of options:

### option 1
- user fills out personal data => submit step 1 of add custom subscription (server)
- user fills out payment details => submit step 2 of add custom subscription (client)
- user clicks submit schedule => 
  - submit step 1 of validation charge (server)
  - submit step 2 of validation charge (client)
  - submit step 3 of validation charge and report errors if any (client)
  - submit step 3 of custom subscription (server)
  - save subscription and saved payment if user selected it (server)
- :tada: 

### option 2
- user fills out personal details => submit add plan and step 1 of transaction + add subscription (server)
- user fills out payment details => submit step 2 of transaction + add subscription (client)
- user clicks submit schedule => submit step 3 of transaction + add subscription (server)
- save data / schedule (server)
- :tada:

This is 100% a hack btw for anyone interested / following along at home. NMI doesn't allow validation of cards while creating a custom subscription.

Option 2 would also allow for validation of a card pretty much out of the box which is really nice. Then we could edit the `Give` block to add support for validating / editing a card which is nice. However it also will probably add a lot more custom plans that are abandoned. We should talk with NMI or find a way to clean these up. Maybe its a job we can run to find plans without subscriptions and delete them if they are older than 1 day?

:+1: option 2 :-1: option 1

**Comment by jbaxleyiii** [comment]
ARRRGGGHHH

...


...


...

adding a subscription does not appear to be possible on a validation request:
```
<?xml version="1.0" encoding="UTF-8"?>
<response>
    <result>3</result>
    <result-text>Invalid Element name add-subscription is not allowed REFID:3179229584</result-text>
    <result-code>300</result-code>
</response>
```

So we are back to option 1 which is WAY messier :-1:

**Comment by jbaxleyiii** [comment]
Fixed in #455
